text/001x-battery_improvements.md

```diff
@@ -0,0 +1,110 @@
+  * Start date: 2022-03-26
+  * Contributors: Hamish Willee <hamishwillee@gmail.com>, ...
+  * Related issues: 
+    - [16-cell support #1747](https://github.com/mavlink/mavlink/pull/1747)
+	- [common: added voltage and current multipliers to BATTERY_STATUS #233](https://github.com/ArduPilot/mavlink/pull/233)
+
+  
+# Summary
+
+This RFC proposes:
+- a new `BATTERY_STATUS_V2` message that has a single cumulative voltage value rather than individual cell voltage arrays.
+- a new (optional) `BATTERY_VOLTAGES` message that can be scaled to as many cells as needed, and which reports them as faults.
+- mechanisms to ease supporting both message types until we can eventually deprecate `BATTERY_STATUS`
+
+  
+# Motivation
+
+The motivation is to:
+- reduce the memory required for battery reporting ([anecdotal evidence](https://github.com/ArduPilot/mavlink/pull/233#issuecomment-976197179) indicates cases with 15% of bandwidth on a default RF channel).
+- Provide a cleaner message and design for both implementers and consumers.
+
+The [BATTERY_STATUS](https://mavlink.io/en/messages/common.html#BATTERY_STATUS) has two arrays that can be used to report individual cell voltages (up to 14), or a single cumulative voltage, and which cannot be truncated.
+The vast majority of consumers are only interested in the total battery voltage, and either sum the battery cells or get the cumulative voltage.
+By separating the cell voltage reporting into a separate message the new battery status can be much smaller, and the cell voltages need only be sent if the consumer is actually interested.
+
+# Detailed Design
+
+There are three parts to the design:
+1. A more efficient status message
+2. A scalable battery voltage message.
+3. Mechanisms that allow the new messages to seamlessly coexist with the old one along with eventual deprecation of the older message.
+
+## BATTERY_STATUS_V2
+
+The message is heavily based on [BATTERY_STATUS](https://mavlink.io/en/messages/common.html#BATTERY_STATUS) but:
+- removes the cell voltage arrays: `voltages` and `voltages_ext`
+- adds `voltage`, the total cell voltage. This is a uint32_t to allow batteries with more than 65V.
+
+
+```xml
+    <message id="???" name="BATTERY_STATUS_V2">
+      <description>Battery information. Updates GCS with flight controller battery status. Smart batteries also use this message, but may additionally send SMART_BATTERY_INFO.</description>
+      <field type="uint8_t" name="id" instance="true">Battery ID</field>
+      <field type="uint8_t" name="battery_function" enum="MAV_BATTERY_FUNCTION">Function of the battery</field>
+      <field type="uint8_t" name="type" enum="MAV_BATTERY_TYPE">Type (chemistry) of the battery</field>
```

Comment:
since both `type` and `function` are provided in SMART_BATTERY_INFO and are non-essential I vote we drop them

Reply:
@dakejahl Originally we had battery-status with everything you needed for working with normal batteries (arguably you don't need type, but you do need function for multi-battery systems). Thus the separation was not just dynamic vs static, but "essential vs optional". 

I think that since we're breaking anyway, we should assume that if a GCS needs the information they WILL read the smart battery info. It's a simpler separation and results in smaller messages/less duplication. So I'll propose removing from V2. Let's see whether that gets through the RFC process.

Reply:
@dakejahl Done, but NOTE, there is a side effect. This means that we are essentially saying that the id field of a battery must always mean the _same_ battery function and chemistry - if you slot in a new battery there will be nothing to tell you that the chemistry has changed or that the battery has changed function. i.e. you would not know that you needed to re-read the SMART_BATTERY_INFO 

I'm not sure whether that matters/how much. If the id maps to a physical connector I would not expect the function to change. However the chemistry changing might matter. 

We might emit `SMART_BATTERY_INFO` on battery change or at very low rate to overcome this.